app/src/main/java/com/rootstrap/android/util/extensions/ActionCallback.kt

```diff
@@ -1,44 +1,57 @@
 package com.rootstrap.android.util.extensions
 
 import com.google.gson.Gson
-import com.rootstrap.android.bus
 import com.rootstrap.android.network.models.ErrorModel
-import com.rootstrap.android.util.ErrorUtil
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import retrofit2.Call
-import retrofit2.Callback
 import retrofit2.Response
 
-abstract class ActionCallback<T> : Callback<T> {
+class ActionCallback {
 
-    override fun onResponse(call: Call<T>, response: Response<T>) {
-        if (response.isSuccessful) {
-            responseAction(response)
-        } else {
-            errorAction(response)
-        }
-    }
+    companion object {
 
-    override fun onFailure(call: Call<T>, t: Throwable) {
-        failureAction(t)
-    }
-
-    open fun responseAction(response: Response<T>) {}
+        suspend fun <T> call(apiCall: Call<T>): Result<Data<T>> =
+            withContext(Dispatchers.IO) {
+                val response = apiCall.execute()
+                manageResponse(response)
+            }
 
-    open fun errorAction(response: Response<T>) {
-        try {
-            response.errorBody()?.let {
-                val error = Gson().fromJson(it.charStream(), ErrorModel::class.java)
-                bus.post(ErrorEvent(ErrorUtil.handleCustomError(error)))
+        private fun <T> manageResponse(response: Response<T>): Result<Data<T>> {
```

Comment:
IMO, `handleResponse` is a more accurate name of what we are doing here. The same would apply for the other `manageSomething` functions.